src/main/java/uk/ac/ebi/ena/webin/cli/service/BiosamplesService.java

```diff
@@ -0,0 +1,106 @@
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.ResourceAccessException;
+import uk.ac.ebi.biosamples.BioSamplesProperties;
+import uk.ac.ebi.biosamples.client.BioSamplesClient;
+import uk.ac.ebi.biosamples.model.Sample;
+import uk.ac.ebi.biosamples.service.AttributeValidator;
+import uk.ac.ebi.biosamples.service.SampleValidator;
+import uk.ac.ebi.ena.webin.cli.service.exception.ServiceException;
+import uk.ac.ebi.ena.webin.cli.utils.RetryUtils;
+
+import java.net.URI;
+import java.util.NoSuchElementException;
+
+public class BiosamplesService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BiosamplesService.class);
+
+    private final BioSamplesClient bioSamplesClient;
+
+    public BiosamplesService() {
+        bioSamplesClient = new BioSamplesClient(
+            URI.create("https://www.ebi.ac.uk/biosamples/"),
+            null,
+            new RestTemplateBuilder(),
+            new SampleValidator(new AttributeValidator()),
+            null,
+            getBioSamplesProperties());
+    }
+
+    public Sample findSampleById(String accession, String webinAuthToken) {
+        if (webinAuthToken == null) {
+            throw new ServiceException("Invalid webin authentication token.");
+        }
+
+        try {
+            return RetryUtils.executeWithRetry(
+                context -> bioSamplesClient.fetchSampleResourceV2(accession, webinAuthToken),
+                context -> LOGGER.warn("Retrying sample retrieval from Biosamples."),
+                HttpServerErrorException.class, ResourceAccessException.class);
+
+        } catch (NoSuchElementException e) {
+            return null;
+        } catch (Exception e) {
+            throw new ServiceException(e, "Could not retrieve BioSample [" + accession + "]");
+        }
+    }
+
+    private BioSamplesProperties getBioSamplesProperties() {
+        return new BioSamplesProperties() {
+            @Override
+            public int getBiosamplesClientPagesize() {
+                return 1000;
+            }
+
+            @Override
+            public int getBiosamplesClientTimeout() {
+                return 60000;
+            }
+
+            @Override
+            public int getBiosamplesClientConnectionCountMax() {
+                return 8;
+            }
+
+            @Override
+            public int getBiosamplesClientThreadCount() {
+                return 1;
+            }
+
+            @Override
+            public int getBiosamplesClientThreadCountMax() {
+                return 8;
+            }
+
+            @Override
+            public int getBiosamplesClientConnectionCountDefault() {
+                return 8;
+            }
+
+            @Override
+            public boolean getAgentSolrStayalive() {
+                return false;
+            }
+
+            @Override
+            public String getOls() {
+                return "https://wwwdev.ebi.ac.uk/ols";
```

Comment:
Extract URL into a static constant:
```
private final static String OLS_PROD_URL = "https://wwwdev.ebi.ac.uk/ols";
```
- Ask Dipayan why do we have a wwwdev service here and if we need this at all.